### PR TITLE
Pass participant as argument in `RoomEvent.ParticipantMetadataChanged`

### PR DESCRIPTION
**Pull request by davidzhao**

### CODE REVIEW

File: src/room/events.ts [hunk 1]
```diff
@@ -194,20 +194,127 @@ export enum RoomEvent {
...
 export enum ParticipantEvent {
+  /**
+   * When a new track is published to room *after* the local
+   * participant has joined. It will not fire for tracks that are already published.
+   *
+   * A track published doesn't mean the participant has subscribed to it. It's
+   * simply reflecting the state of the room.
+   *
+   * args: ([[RemoteTrackPublication]])
+   */
```
Comment (noahlt) [inline src/room/events.ts:205]: 😍

File: src/room/events.ts [hunk 2]
```diff
@@ -194,20 +194,127 @@ export enum RoomEvent {
...
+   * Data received from this participant as sender.
+   * Data packets provides the ability to use LiveKit to send/receive arbitrary payloads.
+   * All participants in the room will receive the messages sent to the room.
+   *
+   * args: (payload: Uint8Array, kind: [[DataPacket_Kind]])
+   */
   DataReceived = 'dataReceived',
+
+  /**
+   * Has apeaking status changed for the current participant
```
Comment (noahlt) [inline src/room/events.ts:303]: typo: "speaking"

File: src/room/Room.ts [hunk 3]
```diff
@@ -170,18 +170,12 @@ class Room extends EventEmitter {
...
-          (metadata: object, p: Participant) => {
-            this.emit(RoomEvent.ParticipantMetadataChanged, metadata, p);
-          },
-        )
+        .on(ParticipantEvent.MetadataChanged, (metadata: object) => {
+          this.emit(RoomEvent.MetadataChanged, metadata, this.localParticipant);
+        })
+        .on(ParticipantEvent.ParticipantMetadataChanged, (metadata: object) => {
+          this.emit(RoomEvent.ParticipantMetadataChanged, metadata, this.localParticipant);
+        })
```
Comment (noahlt) [inline src/room/Room.ts:178]: I'm assuming this is a bugfix and that a previous commit changed the `emit` call to remove Participant from the event arguments?

**Review by noahlt — APPROVED** [review]
typo & question

File: src/room/Room.ts (hunk 3 above)
Comment (davidzhao) [reply src/room/Room.ts:178]: ParticipantEvents don't come with the participant itself, hence we were passing in undefined previously.

File: src/room/events.ts (hunk 2 above)
Comment (davidzhao) [reply src/room/events.ts:303]: thank you!

**Pull request merged**